worldengine/draw.py

```diff
@@ -306,24 +284,16 @@ def draw_precipitation(world, target, black_and_white=False):
     height = world.height
 
     if black_and_white:
-        low = None
-        high = None
-        for y in range(height):
-            for x in range(width):
-                p = world.precipitations_at((x, y))
-                if low is None or p < low:
-                    low = p
-                if high is None or p > high:
-                    high = p
+        low = world.precipitation['data'].min()
+        high = world.precipitation['data'].max()
+        floor = 0
+        ceiling = 255
+
+        colors = numpy.interp(world.precipitation['data'], [low, high], [floor, ceiling])
```

Comment:
I am making use of numpys interp-method (which basically does what drawing_functions.py/gradient() did before). I assume it is faster.
I also prepared things for the 16 Bit grayscale-output that can also be used for temperature and precipitation, if wanted.